src/textfile_gui.cpp

```diff
@@ -654,54 +654,38 @@ void TextfileWindow::ScrollToLine(size_t line)
 /**
  * Do an in-memory gunzip operation. This works on a raw deflate stream,
  * or a file with gzip or zlib header.
- * @param bufp  A pointer to a buffer containing the input data. This
- *              buffer will be freed and replaced by a buffer containing
- *              the uncompressed data.
- * @param sizep A pointer to the buffer size. Before the call, the value
- *              pointed to should contain the size of the input buffer.
- *              After the call, it contains the size of the uncompressed
- *              data.
+ * @param input Buffer containing the input data.
+ * @return Decompressed buffer.
  *
- * When decompressing fails, *bufp is set to nullptr and *sizep to 0. The
- * compressed buffer passed in is still freed in this case.
+ * When decompressing fails, an empty buffer is returned.
  */
-static void Gunzip(uint8_t **bufp, size_t *sizep)
+static std::vector<char> Gunzip(std::vector<char> &input)
```

Comment:
I think `std::span<char>` would be better for the input parameter for these two functions.